### DISCUSSION TITLE
Go Getting github.com/anacrolix/torrent

### DISCUSSION THREAD

**Discussion by yonatanbeb** [anacrolix/torrent#513] [Q&A] [answered]
Hello,
Me and my team are trying to install anacrolix/torrent ( go get github.com/anacrolix/torrent ). On two of our computers, one on a linux VM and the other on windows we get the same errors. On both computers the Go version is **go1.15.7** The errors are as follows:

_**# github.com/anacrolix/missinggo/prioritybitmap**
../../../github.com/anacrolix/missinggo/prioritybitmap/prioritybitmap.go:141:15: cannot use func literal (type func(int) bool) as type func(uint32) bool in argument to me.IterTyped
../../../github.com/anacrolix/missinggo/prioritybitmap/prioritybitmap.go:151:12: cannot use i (type int) as type uint32 in argument to _f
**# github.com/pion/srtp**
../../../github.com/pion/srtp/session_srtp.go:144:9: assignment mismatch: 1 variable but h.Unmarshal returns 2 values
../../../github.com/pion/srtp/srtp.go:37:9: assignment mismatch: 1 variable but header.Unmarshal returns 2 values
../../../github.com/pion/srtp/srtp.go:52:9: assignment mismatch: 1 variable but header.Unmarshal returns 2 values
../../../github.com/pion/srtp/srtp.go:56:51: header.PayloadOffset undefined (type *rtp.Header has no field or method PayloadOffset)
../../../github.com/pion/srtp/srtp_cipher_aead_aes_gcm.go:99:13: header.PayloadOffset undefined (type *rtp.Header has no field or method PayloadOffset)
../../../github.com/pion/srtp/srtp_cipher_aead_aes_gcm.go:104:18: header.PayloadOffset undefined (type *rtp.Header has no field or method PayloadOffset)
../../../github.com/pion/srtp/srtp_cipher_aes_cm_hmac_sha1.go:121:30: header.PayloadOffset undefined (type *rtp.Header has no field or method PayloadOffset)
../../../github.com/pion/srtp/srtp_cipher_aes_cm_hmac_sha1.go:126:32: header.PayloadOffset undefined (type *rtp.Header has no field or method PayloadOffset)
../../../github.com/pion/srtp/stream_srtp.go:88:6: assignment mismatch: 1 variable but header.Unmarshal returns 2 values_

They occur in the two packages above -- anacrolix/missinggo/prioritybitmap & pion/srtp. My team has looked into one of the errors in pion/srtp and have found that these errors exist in all released tags. The error we found in both tags is that the Header struct in pion/rtp has an Unmarshal function which returns two values, and in srtp it is assigned to a single value.
We have tried updating the package many times and in various combinations.
We are sure that the problem we are facing is a silly one since so many of you have been able to install and use this library. Is any one familiar with the problems we are facing?
Many thanks in advance from the brattle team!

**Comment by anacrolix** [comment, marked as answer]
Are you using Go modules?

**Reply by anacrolix** [reply]
It's highly likely the OP is not using Go modules, which isn't supported.

**Comment by anacrolix** [comment]
You might try this again with a more recent version of anacrolix/torrent. There has been some updates made to the deps in go.mod that might more closely match usage without modules.
